dotnet/src/Functions/Functions.OpenApi.Extensions/Extensions/DeclarativeAgentExtensions.cs

```diff
@@ -0,0 +1,131 @@
+        var logger = loggerFactory.CreateLogger(typeof(DeclarativeAgentExtensions)) ?? NullLogger.Instance;
+        using var declarativeAgentFileJsonContents = DocumentLoader.LoadDocumentFromFilePathAsStream(filePath,
+            logger);
+
+        var results = await DCManifestDocument.LoadAsync(declarativeAgentFileJsonContents, new ReaderOptions
+        {
+            ValidationRules = [] // Disable validation rules
+        }).ConfigureAwait(false);
+
+        if (!results.IsValid)
+        {
+            var messages = results.Problems.Select(static p => p.Message).Aggregate(static (a, b) => $"{a}, {b}");
+            throw new InvalidOperationException($"Error loading the manifest: {messages}");
+        }
+
+        var document = results.Document ?? throw new InvalidOperationException("Error loading the manifest");
+        var manifestDirectory = Path.GetDirectoryName(filePath);
+        document.Instructions = await GetEffectiveInstructionsAsync(manifestDirectory, document.Instructions, logger, cancellationToken).ConfigureAwait(false);
+
+        var agent = new T
+        {
+            Name = document.Name,
+            Instructions = document.Instructions,
+            Kernel = kernel,
+            Arguments = new KernelArguments(promptExecutionSettings ?? new PromptExecutionSettings()
+            {
+                FunctionChoiceBehavior = FunctionChoiceBehavior.Auto(),
+            }),
+            Description = document.Description,
+            LoggerFactory = loggerFactory,
+            Id = string.IsNullOrEmpty(document.Id) ? Guid.NewGuid().ToString() : document.Id!,
+        };
+
+        if (document.Capabilities is { Count: > 0 })
+        {
+            logger.LogWarning("Importing capabilities from declarative agent is not supported in semantic kernel.");
+        }
+
+        if (document.Actions is { Count: > 0 })
+        {
+            logger.LogInformation("Importing {ActionsCount} actions from declarative agent.", document.Actions.Count);
+            await Task.WhenAll(document.Actions.Select(action => ImportCAPFromActionAsync(action, manifestDirectory, kernel, pluginParameters, logger, cancellationToken))).ConfigureAwait(false);
+        }
+        return agent;
+    }
+    private static async Task ImportCAPFromActionAsync(DCAction action, string? manifestDirectory, Kernel kernel, CopilotAgentPluginParameters? pluginParameters, ILogger logger, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var capManifestPath = GetFullPath(manifestDirectory, action.File);
+            logger.LogInformation("Importing action {ActionName} from declarative agent from path {Path}.", action.Id, capManifestPath);
+            await kernel.ImportPluginFromCopilotAgentPluginAsync(action.Id, capManifestPath, pluginParameters, cancellationToken).ConfigureAwait(false);
+        }
+        catch (Exception ex) when (ex is FileNotFoundException or InvalidOperationException)
```

Comment:
nit: This catch block with the condition is the same as the generic one below. Either remove this catch block or add special handling that is different from the other one.